src/service/service.rs

```diff
@@ -0,0 +1,165 @@
+use std::error::Error as StdError;
+use std::fmt;
+use std::marker::PhantomData;
+
+use futures::{future, Future, IntoFuture};
+
+use body::Payload;
+use common::Never;
+use ::{Request, Response};
+
+/// An asynchronous function from `Request` to `Response`.
+pub trait Service {
+    /// The `Payload` body of the `http::Request`.
+    type ReqBody: Payload;
```

Comment:
Doesn't Hyper control what type the request body is?

Reply:
True. I'd like to bring the implementation back for `Client`, which will need to define the `ReqBody`. I'm just unsure about how best to move forward to `tower::Service`, which has a `poll_ready` function (and [may be changing soon](https://github.com/tower-rs/tower/issues/66))...